src/ApiException.php

```diff
@@ -78,6 +78,49 @@ public function getStatus()
         return $this->status;
     }
 
+    // Returns null if metadata does not contain error info, or return containsErrorInfo() array
+    // if the metadata does contain error info.
+    public static function decodeMetadataErrorInfo($metadata)
+    {
+        $details = [];
+        // ApiExceptions created from RPC status have metadata that is an array of objects.
+        if (is_object(reset($metadata))) {
```

Comment:
What are you trying to do by calling `reset`? Seems to me manipulating the array's internal pointer is probably not a great idea ([docs](https://www.php.net/manual/en/function.reset.php)).

Reply:
I was just trying to get the first element of the array; so to your point, I don't actually need the internal pointer to be changed. I could update this to `is_object($metadata[array_key_first($metadata)])` (and similar for line 93) - unless there's a better way to check this?

`Can you add php doc comments to all of the added public APIs?` - sorry, what does that mean?

Reply:
So what is the type of first element of the array for gRPC metadata? It isn't an object?

Personally I think in this case, `reset` is better. 

I added an example above wrt PHPDoc